dependencies.yaml

```diff
@@ -509,7 +513,7 @@ dependencies:
           - seaborn
           - *scikit_learn
           - statsmodels
-          - umap-learn==0.5.3
+          - umap-learn==0.5.6
```

Comment:
`umap-learn==0.5.3` does not have Python 3.12 packages on `conda-forge`, resulting in these build errors:

```text
  LibMambaUnsatisfiableError: Encountered problems while solving:
    - package umap-learn-0.5.3-py310hff52083_0 requires python_abi 3.10.* *_cp310, but none of the providers can be installed
  
  Could not solve for environment specs
  The following packages are incompatible
  ...
  └─ umap-learn 0.5.3  is not installable because there are no viable options
     ├─ umap-learn 0.5.3 would require
     │  └─ python_abi 3.10.* *_cp310, which conflicts with any installable versions previously reported;
     ├─ umap-learn 0.5.3 would require
     │  └─ python_abi 3.7.* *_cp37m, which conflicts with any installable versions previously reported;
     ├─ umap-learn 0.5.3 would require
     │  └─ python_abi 3.8.* *_cp38, which conflicts with any installable versions previously reported;
     ├─ umap-learn 0.5.3 would require
     │  └─ python_abi 3.9.* *_cp39, which conflicts with any installable versions previously reported;
     └─ umap-learn 0.5.3 would require
        └─ python_abi 3.11.* *_cp311, which conflicts with any installable versions previously reported.
```

([build link](https://github.com/rapidsai/cuml/actions/runs/10710717433/job/29698022362?pr=6060#step:7:971))

Proposing updating to the latest version (0.5.6).